nni/contrib/compression/utils/evaluator.py

```diff
@@ -1104,3 +1148,363 @@ def evaluate(self) -> Tuple[float | None, Dict[str, Any]]:
 
     def get_dummy_input(self) -> Any:
         return self.dummy_input
+
+
+class DeepspeedTorchEvaluator(Evaluator):
+    def __init__(self, training_func: _TRAINING_FUNC, optimizer: Optimizer | Callable[[List[Tensor]], Optimizer] | None,
+                 training_step: _TRAINING_STEP, lr_scheduler: SCHEDULER | Callable[[Optimizer], SCHEDULER] | None,
```

Comment:
if `optimizer` and `lr_scheduler` can be wrong, I think we could give them a default `None` value.